conclave-host/src/main/kotlin/com/r3/conclave/host/internal/NativeEnclaveHandle.kt

```diff
@@ -15,21 +22,57 @@ class NativeEnclaveHandle(
 ) : EnclaveHandle {
     private val enclaveFile: Path
     private val enclaveId: Long
-    override val enclaveInterface: HostEnclaveInterface
+    override val enclaveInterface: NativeHostEnclaveInterface
+    override lateinit var quotingService: EnclaveQuoteService
+
+    /**
+     * Handler for servicing requests from the enclave for signed quotes.
+     */
+    private inner class GetSignedQuoteHandler : CallHandler {
+        override fun handleCall(parameterBuffer: ByteBuffer): ByteBuffer {
+            val report = Cursor.slice(SgxReport, parameterBuffer)
+            val signedQuote = quotingService.retrieveQuote(report)
+            return signedQuote.buffer
+        }
+    }
 
     init {
-        require(enclaveMode != EnclaveMode.MOCK)
+        require(enclaveMode != EnclaveMode.MOCK) {
+            MOCK_MODE_UNSUPPORTED_MESSAGE
+        }
+
         NativeLoader.loadHostLibraries(enclaveMode)
         enclaveFile = Files.createTempFile(enclaveClassName, "signed.so").toAbsolutePath()
         enclaveFileUrl.openStream().use { Files.copy(it, enclaveFile, REPLACE_EXISTING) }
         enclaveId = Native.createEnclave(enclaveFile.toString(), enclaveMode != EnclaveMode.RELEASE)
         enclaveInterface = NativeHostEnclaveInterface(enclaveId)
         NativeApi.registerHostEnclaveInterface(enclaveId, enclaveInterface)
+
+        enclaveInterface.apply {
+            registerCallHandler(HostCallType.GET_SIGNED_QUOTE, GetSignedQuoteHandler())
+        }
     }
 
-    override fun initialise() {
+    override fun initialise(attestationParameters: AttestationParameters?) {
         synchronized(this) {
-            enclaveInterface.initializeEnclave(enclaveClassName)
+            quotingService = getQuotingService(attestationParameters)
+            initializeEnclave(enclaveClassName)
+        }
+    }
+
+    /** Get the appropriate quoting service. */
+    private fun getQuotingService(attestationParameters: AttestationParameters?): EnclaveQuoteService {
```

Comment:
Maybe not strictly related to this PR, but being this function called basically only once, is it worth to put it inside the companion object? Wondering if there's a Java/Kotlin convention for this.

Reply:
Could do that, but we would have to pass the enclave mode as a parameter too.